src/core/DataFile.cpp

```diff
@@ -1638,6 +1639,51 @@ void DataFile::upgrade_1_3_0()
 	}
 }
 
+void DataFile::upgrade_noHiddenAutomationTracks()
+{
+	// convert global automation tracks to non-hidden
+	QDomElement song = firstChildElement("lmms-project")
+		.firstChildElement("song");
+	QDomElement tc = song.firstChildElement("trackcontainer");
+	QDomElement gaTrack = song.firstChildElement("track");
+	if (!gaTrack.isNull()
+		&& gaTrack.attribute("type").toInt() == static_cast<int>(Track::Type::HiddenAutomation))
+	{
+		// global automationclips
+		QDomNodeList aps = gaTrack.elementsByTagName("automationclip");
+		QList<QDomNode> tracksToInsert;
+		for (int i = 0; i < aps.length(); ++i)
+		{
+			QDomElement ap = aps.item(i).toElement();
```

Comment:
Similar as above. Would be better to understand if `ap` was called `automationClip`.